community/kernel/src/main/java/org/neo4j/kernel/impl/proc/ProcedureRegistry.java

```diff
@@ -37,11 +37,11 @@
     private final Map<ProcedureSignature.ProcedureName,CallableProcedure> procedures = new HashMap<>();
 
     /**
-     * Register a new procedure. This method must not be called concurrently with {@link #get(ProcedureSignature.ProcedureName)}.
+     * Register a new procedure.
      *
      * @param proc the procedure.
      */
-    public synchronized void register( CallableProcedure proc ) throws ProcedureException
+    public void register( CallableProcedure proc ) throws ProcedureException
```

Comment:
from what i can see we do not need any sync at all here; switching them to CHM instead is a fallback step :)